Don't sign GET or HEAD requests

lukehutch opened the pull request:
Don't sign GET or HEAD requests to GCP Storage. Fixes #2517.

## Pre-launch Checklist

- [x] I read the [Contribute](https://docs.serverpod.dev/contribute) page and followed the process outlined there for submitting PRs.
- [x] This update contains only one single feature or bug fix and nothing else. (If you are submitting multiple fixes, please make multiple PRs.)
- [x] I read and followed the [Dart Style Guide](https://dart.dev/guides/language/effective-dart/style) and formatted the code with [dart format](https://dart.dev/tools/dart-format).
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`), and made sure that the documentation follows the same style as other Serverpod documentation. I checked spelling and grammar.
- [ ] I added new tests to check the change I am making.
- [ ] All existing and new tests are passing.
- [x] Any breaking changes are documented below.

vlidholt commented:
I will check if this really is broken, as we have a client that I know for sure is currently using this function in production on GCP. I tried it out a few months ago and then everything was working correctly.

This PR removes a layer of security and will not work with the private file storage public. The private file storage allows users to upload files to a bucket that is only accessible from Serverpod and not exposed publicly on the web.

lukehutch commented:
Right, and obviously the signing needs to work or files cannot be deleted.

It's definitely broken on my deployment. What could break in the signing process?

vlidholt commented:
@lukehutch, are you sure you are using the right keys in your passwords file?

lukehutch commented:
@vlidholt Yes, as I said in #2517:

> I verified my `HMACAccessKeyId` and `HMACSecretKey` are correct. I even re-generated these, replaced them in passwords.yaml, updated the GitHub secrets, re-deployed my server, and yet this didn't fix the problem.
